src/table.jl

```diff
@@ -28,6 +28,20 @@ function Table{names}(nt::NamedTuple{names2, <:Tuple{Vararg{AbstractVector}}}, i
     end
 end
 
+# TableTraits.jl integration
+function Table(source)
+    iit = TableTraits.isiterabletable(source)
+    if iit===true
+        cols, names = TableTraitsUtils.create_columns_from_iterabletable(source, na_representation=:datavalue)
```

Comment:
One more comment about this: you could also call this with ``na_representation=:missing``, and then any column with missing would be a ``Vector{Union{T,Missing}}``. With the current code version any column with missing you would get as a ``DataValueVector{T}``.